### DISCUSSION TITLE
HTTP Cache of requests

### DISCUSSION THREAD

**Discussion by vlados** [livewire/livewire#4579] [3. Feature ideas]
For one of my projects I started implementing HTTP Cache (If the content is not modified since date, return status 304 with no content).
https://github.com/vlados/vladko.dev/blob/e123a498512cabe4b2b992368c5e116278689573/app/Http/Controllers/HomeController.php#L92
```php
        return (new SetCacheHeaders())->handle(new Request(), function () use ($tags, $lastModifiedDate, $projects, $faq) {
            $response = Response::view('welcome', [
                'faq' => $faq,
                'projects' => $projects,
                'project_tags' => $tags,
            ]);
            $response->setPublic();
            $response->setLastModified($lastModifiedDate);

            $response->header('Last-modified', $lastModifiedDate);
            if ($response->isNotModified(\request())) {
                $response->setContent(null);
                $response->setStatusCode(304);
            }

            return $response;
        }, 'public;etag;max_age=10800;last_modified=' . $lastModifiedDate);
```
This can make the requests, that are not making any changes to be ignored, making Livewire even faster
I am still trying to make it work because there is no documentation for this but I think it will be nice idea?

### **What you think?**

**Comment by joshhanley** [comment]
@vlados Livewire already has a hash of the html, so it can detect if anything has changed or not, and if it hasn't it doesn't send any new html in the response. Is that what you are after?

**Reply by vlados** [reply]
partially, in that case its returning http status 200 with some response data,
It can return http status 304 - not modified without any response data which can speed up this requests
